### ISSUE TITLE
Error if answer has ? 

### ISSUE THREAD

**Issue by neeesn** [marcusgreen/moodle-qtype_gapfill#101]
**Describe the bug**
If theres questionmark in the question, the answer is returned wrong even if the question is right. 

**To Reproduce**
1. Make a drag question where the answer is with an ? 
2. make a preview and click "Fill in correct responses" and then "Submit and finish."
3. It shows errors in the correct answers

**Expected behavior**
Show correct answer even with question mark

![image](https://user-images.githubusercontent.com/1931926/216272081-111908d7-0719-4476-be6e-366022175ea2.png)

**Desktop (please complete the following information):**
  - Browser  chrome, Version 109.0.5414.120 (Official Build) (64-bit)
 - Moodle [3.11.9 (Build: 20220822)](http://docs.moodle.org/dev/Releases)
 - Plugin 2.134 (2022061900)

**Comment by marcusgreen** [comment]
Hi Neesn, thank you for your detailed report. 
Can you confirm the status of the (slightly hidden) Disable Regex box. 
See https://docs.moodle.org/401/en/Gapfill_question_type#Disable_Regex
I just got this to work 
The [?cat] sat on the [mat?]

**Comment by neeesn** [comment]
![image](https://user-images.githubusercontent.com/1931926/216569270-4a60cce7-36c0-4b04-b745-e2db346f9226.png)

**Comment by neeesn** [comment]
It is not disabled

**Comment by marcusgreen** [comment]
It needs to be disabled (checked) if you don't want the characters to be interpreted as regular expressions. I think it could be clearer.

**Comment by neeesn** [comment]
Super that seems to be working :-) 
But you are right - a better description would be nice. Most teachers dont know what regex is.

**Comment by marcusgreen** [comment]
You are right, very few people will know will know what regex is.  It can be disabled at a system wide level (I know if you are a teacher that may mean nothing). I was thinking some more about that today. A few years ago I changed it so that by default regex is disabled, but it would be good if whenever it is on a clear sign is displayed on the question editing page saying something like "Regex processing is enabled, characters like ?* etc will have a special meaning, do you want that".

Can I mention you in the update document if/when I update that, as I like to credit anyone who has given me feedback on how it works. Is there a longer version of your name, or shall I put neesn?

**Comment by dinism** [comment]
Thanks a lot neeesn (Henrik Sune Pedersen) and marcusgreen (Marcus Green).
